doc/src/build/move/sui-move-library.md

```diff
@@ -23,54 +23,43 @@ transfer::transfer(obj, recipient);
-To transfer an object `obj` (whose owner is an address) to another object `owner`:
-```
-transfer::transfer_to_object(obj, &mut owner);
-```
-This function returns a `ChildRef` instance that cannot be dropped arbitrarily. It can be stored in the parent as a field.
-Sometimes we need to set the child field of a parent while constructing it. In this case, we don't yet have a parent object to transfer into. In this case, we can call the `transfer_to_object_id` API. Example:
-```
-let parent_info = object::new(ctx);
-let child = Child { info: object::new(ctx) };
-let (parent_id, child_ref) = transfer::transfer_to_object_id(child, parent_info);
-let parent = Parent {
-    info: parent_info,
-    child: child_ref,
-};
-transfer::transfer(parent, tx_context::sender(ctx));
-```
-To transfer an object `child` from one parent object to a new parent object `new_parent`, we can use the following API:
 ```
-transfer::transfer_child_to_object(child, child_ref, &mut new_parent);
+use sui::dynamic_object_field as ofield;
+
+let a: &A = /* ... */;
+let b: &B = /* ... */;
+
+// Adds `b` as a dynamic object field to `a` with "name" `0: u8`.
+ofield::add<u8, B>(&mut a.id, 0, b);
 ```
-Note that in this call, we must also have the `child_ref` to prove the original ownership. The call will return a new instance of `ChildRef` that the new parent can maintain.
-To transfer an object `child` from an object to an address `recipient`, we can use the following API:
+
+Once an object is owned by another object, it is required that any references to that object go through its owner: If an object-owned object is passed as an input to an entry function in a transaction, that transaction will fail. For instance, if we have a chain of ownership: address `Addr1` owns object `a`, object `a` owns object `b`, and `b` owns object `c` then in order to use object `c` in a Move call, the transaction must be signed by `Addr1`, and accept `a` as an input, and `b` and `c` must be accessed dynamically during transaction execution:
+
 ```
-transfer::transfer_child_to_address(child, child_ref, recipient);
+use sui::dynamic_object_field as ofield;
+
+// signed of ctx is Addr1
+public entry fun entry_function(a: &A, ctx: &mut TxContext) {
+  let b: &B = ofield::borrow<u8, B>(&a.id, 0);
+  let c: &C = ofield::borrow<u8, C>(&b.id, 0);
+}
 ```
-This call also requires to have the `child_ref` as proof of original ownership.
-After this transfer, the object will be owned by `recipient`.
 
 More examples of how objects can be transferred and owned can be found in
 [object_owner.move](https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/unit_tests/data/object_owner/sources/object_owner.move).
 
-### Freeze an object
+### Shared and immutable
```

Comment:
```suggestion
### Immutable
```

Reply:
Some real old terminology here, and we might want to fix some things below. We used to refer to immutable objects as shared immutable and shared objects as shared mutable. 
But we now just focus on the important access patterns for those, so "immutable" and "shared" objects.